### DISCUSSION TITLE
[Feature request] `set-cookie` scriptlet

### DISCUSSION THREAD

**Discussion by scripthunter7** [uBlockOrigin/uBlock-issues#2352] [Ideas] [answered]
As far as I can see, there is a [cookie-remover](https://github.com/gorhill/uBlock/wiki/Resources-Library#cookie-removerjs-) scriptlet, but there is no way to override a cookie (but correct me if I'm wrong).

There are several pages that shows aggressive popups (which is block scrolling), and displaying of these popups usually depends on 1 cookie. For example, egeszsegkalauz.hu shows popup this way:

<details>

<summary>Show JS</summary>

```js
$(window).ready(() => {
    if (!getCookie('isLeavingPopUpReady') && document.querySelector('.leavingPopupLayer')) {
        console.log("first");
        const showPopup = () => {
            setCookie('isLeavingPopUpReady', false, 3);
            document.querySelector('body').style.overflow = 'hidden';
            document.querySelector('.leavingPopupLayer').style.display = 'block';
            document.querySelector('.leavingPopupLayer div').classList.add('fade-in');
        };

        // ...
    }
});
```

</details>

It would be much easier to block these popups if cookie overwrite was available: 
```adblock
egeszsegkalauz.hu##+js(set-cookie, isLeavingPopUpReady, 1)
```

For security reasons, `set-cookie` should only take predefined values, just like [set-constant](https://github.com/gorhill/uBlock/wiki/Resources-Library#set-constantjs-).

AdGuard has a set-cookie scriptlet, reference: https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-scriptlets.md#set-cookie

**Comment by uBlock-user** [comment]
duplicate of https://github.com/uBlockOrigin/uBlock-issues/issues/1518

**Reply by scripthunter7** [reply]
Hmm, right, sorry. Although it's been more than 1 year since then. As far as I know, this scriptlet works without problems in AdGuard.

**Reply by peace2000** [reply, marked as answer]
@scripthunter7 Set cookie is added recently https://github.com/gorhill/uBlock/releases
